mitosheet/mitosheet/step_performers/graph/graph.py

```diff
@@ -0,0 +1,201 @@
+
+        height = (
+            graph_rendering["height"] if "height" in graph_rendering_keys else "100%"
+        )
+        width = graph_rendering["width"] if "width" in graph_rendering_keys else "100%"
+
+        # Create a copy of the dataframe, just for safety.
+        df: pd.DataFrame = prev_state.dfs[sheet_index].copy()
+        df_name: str = prev_state.df_names[sheet_index]
+
+        if len(x_axis_column_ids) == 0 and len(y_axis_column_ids) == 0:
+            fig = ''
+            html_and_script = {'html': '', 'script': ''}
+            graph_generation_code = ''
+        else: 
+            fig = get_plotly_express_graph(
+                graph_type,
+                df,
+                safety_filter_turned_on_by_user,
+                x_axis_column_headers,
+                y_axis_column_headers,
+            )
+
+            # Get rid of some of the default white space
+            fig.update_layout(
+                margin=dict(
+                    l=0,
+                    r=0,
+                    t=30,
+                    b=30,
+                )
+            )
+
+            html_and_script = get_html_and_script_from_figure(fig, height, width)
+
+            graph_generation_code = get_plotly_express_graph_code(
+                graph_type,
+                df,
+                safety_filter_turned_on_by_user,
+                x_axis_column_headers,
+                y_axis_column_headers,
+                df_name,
+            )
+
+        post_state.graph_data_json[str(sheet_index)] = {
+            "graphParams": {
+                "graphPreprocessing": graph_preprocessing,
+                "graphCreation": graph_creation,
+                "graphStyling": {},
+                "graphRendering": graph_rendering,
+            },
+            "graphGeneratedCode": graph_generation_code,
+            "graphHTML": html_and_script["html"],
+            "graphScript": html_and_script["script"],
+        }
+
+        return post_state, None
+
+    @classmethod
+    def transpile(  # type: ignore
+        cls,
+        prev_state: State,
+        post_state: State,
+        execution_data: Optional[Dict[str, Any]],
+        graph_preprocessing: Any,
+        graph_creation: Any,
+        graph_rendering: Any,
+        **params,
+    ) -> List[str]:
+        return []
+
+    @classmethod
+    def describe(  # type: ignore
+        cls,
+        graph_preprocessing: Any,
+        graph_creation: Any,
+        graph_rendering: Any,
+        df_names=None,
+        **params,
+    ) -> str:
+        sheet_index = graph_creation["sheet_index"]
+        graph_type = graph_creation["graph_type"]
+        if df_names is not None:
+            df_name = df_names[sheet_index]
+            return f"Graphed {df_name} as {GRAPH_TITLE_LABELS[graph_type]}"
+        return f"Created {GRAPH_TITLE_LABELS[graph_type]}"
+
+    @classmethod
+    def get_modified_dataframe_indexes(cls, graph_creation, **params) -> Set[int]:  # type: ignore
```

Comment:
@naterush what do we want to do here? 

If it returns an empty set, then this step modified every dataframe. If it returned -1, then it modified all new dataframes (on the left side of the dfs array).

Neither of which we want to do

Reply:
I think -1 is fine, since it doesn't create any new data frames we aren't actually modifying anything. 

That being said, we'll have to keep an eye on this this effects other optimization efforts...